libvisual-plugins/plugins/actor/infinite/display.c

```diff
@@ -162,15 +162,28 @@ static void _inf_compute_surface(InfinitePrivate *priv, t_interpol* vector_field
 		for (j=0;j<priv->plugheight;j++) {
 			for (i=0;i<priv->plugwidth;i++) {
 				interpol = &vector_field[add_dest];
-				add_src = (interpol->coord & 0xffff) * priv->plugwidth + (interpol->coord >> 16);
-				ptr_pix = priv->surface1 + add_src;;
+				const uint16_t y = interpol->coord & 0xffff;
+				const uint16_t x = interpol->coord >> 16;
+				(void)add_src;
+				ptr_pix = priv->surface1 + y * priv->plugwidth + x;
 
-				/* FIXME it does buffer overread here now and then */
+				const uint8_t *ptr_pix_end = priv->surface1 + (priv->plugwidth * priv->plugheight);
 
```

Comment:
@hartwork Not sure if I missed this for master, but this line should be hoisted out of the rendering loop. `ptr_pix_end` depends only on the bitmap data pointer and dimensions, which does not change in the loop.

The compiler should be able to easily optimize loop invariants like these, but I think it makes sense organizationally to have the computation done before the loop. It also makes debug builds faster.

Reply:
@kaixiong alright, done and pushed. I'll backport to 0.4.x after the merg.